vignettes/pull_request_comments.Rmd

```diff
@@ -0,0 +1,249 @@
+mutually exclusive, meaning the comments from each endpoint do not
+appear in other endpoints.
+
+```{r Collect comments by pull request number}
+#gh call but with date
+# get the data
+gh_response_issue_or_pr_comment <- github_api_project_issue_or_pr_comment_refresh(owner,repo,token,save_path_issue_or_pr_comments,verbose=TRUE)
+
+# create directory and iterate over data
+#dir.create(save_path_issue_or_pr_comments)
+github_api_iterate_pages(token,gh_response_issue_or_pr_comment,
+                         save_path_issue_or_pr_comments,
+                         prefix="issue_or_pr_comment",
+                         verbose=TRUE)
+```
+
+```{r}
+all_issue_or_pr_comments <- lapply(list.files(save_path_issue_or_pr_comments,
+                                     full.names = TRUE),read_json)
+all_issue_or_pr_comments <- lapply(all_issue_or_pr_comments,
+                                   github_parse_project_issue_or_pr_comments)
+all_issue_or_pr_comments <- rbindlist(all_issue_or_pr_comments,fill=TRUE)
+
+head(all_issue_or_pr_comments,1000)  %>%
+  gt(auto_align = FALSE) 
+```
+
+## Pull Request Review Endpoint
+
+We can obtain information on the whole review itself in a pull request,
+which is different than an in-line comment on code review. The review
+endpoint allows analysts to see the state of the review such as
+requesting changes, approving, or a singular comment. More information
+on the fields can be seen on the function definition for
+`github_parse_project_pr_reviews`. These fields of data can greatly
+contribute to finding and analyzing patterns in collaboration between
+developers.
+
+```{r Collect Reviews from Pull Requests}
+gh_response <- github_api_pr_reviews_refresh(owner, repo, pull_number = 330, token, save_path_pr_reviews)
+
+file_name <- paste0(save_path_pr_reviews,
+                    owner,"_",repo,"_pullreview",
+                    ".json")
+#write to json.
+write_json(gh_response,file_name,pretty=TRUE,auto_unbox=TRUE)
+message("Written to file: ", file_name)
+```
+
+```{r Parse Review Comments from Pull Requests}
+all_pr_reviews <- lapply(list.files(save_path_pr_reviews, full.names = TRUE), read_json)
+all_pr_reviews <- lapply(all_pr_reviews, github_parse_project_pr_reviews)
+all_pr_reviews <- rbindlist(all_pr_reviews, fill = TRUE)
+head(all_pr_reviews,2)  %>%
+  gt(auto_align = FALSE) 
+```
+
+## Pull Request Comments Endpoint
+
+Other information we can obtain is the review comments from Pull
+Requests that contain the inline code blocks. The review comments
+include the diff_hunk field, which is a string that contains the in-line
+code reviewers may comment on the pull request. The particular line
+numbers the comment is referencing is specified by the review comment.
+More information on the fields can be seen on the function definition
+for `github_parse_project_pr_comments`. These fields of data can greatly
+contribute to finding and analyzing patterns in collaboration between
+developers.
```

Comment:
I would mention briefly here the start line and end line information with an actual example in the Notebook. Make sure you subset a table by the PR id in kaiaulu and explain it, maybe linking to the browser side.